kyo-core/shared/src/main/scala/kyo/logs.scala

```diff
@@ -86,110 +66,45 @@ object Logs:
             inline def errorEnabled: Boolean = logger.isErrorEnabled
 
             inline def trace(msg: => String)(
```

Comment:
Can you make `msg` and `file` also `inline` in these methods? With this change, the message won't even be evaluated/allocated if the logging level isn't enabled. Inlining `file` doesn't avoid allocations or evaluation but it saves some byte codes.

Reply:
From my initially testing, this is not possible. I can try experimenting with an abstract class, though not sure if that will change anything.

```
Coordinator.scala:68:16 
[error] 68 |                Logs.unsafe.error("Kyo coordinator failure", ex)
[error]    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |          Deferred inline method error in trait Unsafe cannot be invoked
```

Inlining method parameters requires the full function to be inlined, and that results in these errors wherever `Unsafe` is used.

Reply:
Even if we switch to a class implementation of SL4J (extending a trait), the inlining will break usage of the local - as the implementation of the trait is not known:

```
[error]  12 |            _ <- Logs.trace("trace")
[error]     |                 ^^^^^^^^^^^^^^^^^^^
[error]     |        Deferred inline method trace in trait Unsafe cannot be invoked
[error]     |---------------------------------------------------------------------------
[error]     |Inline stack trace
[error]     |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]     |This location contains code that was inlined from logs.scala:123
[error] 123 |        logWhen(_.traceEnabled)(_.trace(msg))
[error]     |                                ^^^^^^^
[error]      ---------------------------------------------------------------------------

```

Reply:
I see. Would it work if `Unsafe`'s methods aren't inlined? Inlining the main API to avoid the evaluation of the message in case the logging is disabled seems more valuable.

Reply:
Currently the methods in `Logs` object are inlined, along with the methods in `Unsafe` on the SL4J instance. We just cannot inline the `Unsafe` function parameters as they are not in the trait.